src/connector.c

```diff
@@ -281,6 +300,32 @@ tlsuv_connector_req proxy_connect(uv_loop_t *loop, const tlsuv_connector_t *self
     return r;
 }
 
+int proxy_set_auth(tlsuv_connector_t *self, tlsuv_auth_t auth, const char *username, const char *password) {
+    struct tlsuv_proxy_connector_s *c = (struct tlsuv_proxy_connector_s *) self;
+    if (auth == tlsuv_PROXY_NONE) {
+        c->auth_header = NULL;
+        free(c->auth_value);
```

Comment:
`tlsuv_new_proxy_connector` allocates the connector with `malloc`. Should it use `calloc` or `memset`, and check for null here (and line 314) before freeing?

Reply:
yes, I change to use `calloc`. free is safe to use on NULL pointers